### DISCUSSION TITLE
Extra Global Label Options

### DISCUSSION THREAD

**Discussion by Maraza25** [octoposprime/op-docs#14] [Ideas]
How about adding the following to the label section?
- feature
- ui/ux
- performance
- security
- research
- deployment
- fix
- style
- config
- test
- sub-issue
- monitoring
- refactor
- critical
- build
- integration

@octoposprime/design
@octoposprime/design-jr
@octoposprime/design-tl
@octoposprime/be
@octoposprime/be-jr
@octoposprime/be-tl
@octoposprime/mo
@octoposprime/mo-jr
@octoposprime/mo-tl
@octoposprime/devops
@octoposprime/devops-jr
@octoposprime/devops-tl
@octoposprime/game
@octoposprime/game-jr
@octoposprime/game-tl
@octoposprime/em
@octoposprime/em-jr
@octoposprime/em-tl
@octoposprime/web
@octoposprime/web-jr
@octoposprime/web-tl
@octoposprime/fe
@octoposprime/fe-jr
@octoposprime/fe-tl
@octoposprime/docs
@octoposprime/docs-jr
@octoposprime/docs-tl
@octoposprime/arc
@octoposprime/arc-jr
@octoposprime/arc-tl

**Comment by husamettinarabaci** [comment]
crictical (#8b0000)
high-priority (#C61700)
medium-priority (#D93F0B)
low-priority (#FDCC21)

**Comment by Sddilora** [comment]
- **change** (Feature changing) (#60AD72)
- **cross team** (Requires collaboration between teams) (#56AB5F)
- **internal feedback** (Identified internally) (#D7478D)
- **not reproducible** (The issue cannot be reproduced on the environments of the team members or team leads.) (#f9d0c4)
- **parent-issue** (This issue has sub-issues) (#595068)
- **refactor** (Improving codebase structure, organization, or performance without altering external behavior) (#BBA612)
- **sub-issue** (This issue has a parent issue) (#118F8E)
- **testing** (Indicates issues or tasks undergoing testing and quality assurance.) (#D3CCF8)
- **user feedback** (Reported by users or stakeholders.) (#9A5F9F)
- **planning** ( Indicates issues or tasks in the planning phase, such as requirements gathering or initial scoping. )
- **development** (Indicates issues or tasks currently in active development.)
- **deployment** ( Indicates issues or tasks related to deployment to production or release.)

**Comment by SoyluAlaattin** [comment]
enhancement - For indicating improvements. (#FFD700)
design - For issues related to design or user interface. (#FFA07A)
scalability - For issues related to scalability. (#20B2AA)
compliance - For specifying compliance requirements. (#9370DB)
analysis - For data analysis or project research. (#00FFFF)
deployment-strategy - For specifying deployment strategies. (#FF6347)
formatting - For issues related to formatting or style. (#6495ED)
configuration - For issues related to configuration files or settings. (#FFA500)
automated-testing - For issues related to automated testing. (#FF4500)
subtask - For specifying subtasks of a main task. (#ADFF2F)
performance-optimization - For performance optimization tasks. (#FF69B4)
system-monitoring - For system monitoring or diagnostics. (#32CD32)
code-refactoring - For code refactoring tasks. (#8A2BE2)
urgent-fix - For critical bugs requiring urgent fixes. (#FFD700)
continuous-integration - For issues related to continuous integration processes. (#800000)

**Comment by yusufyilmazf** [comment]
- **routing**: PCB signal paths - routing or editing (#3498db)
- **layout**: PCB layout related (#2ecc71)
- **footprint**: Component footprint management (#e67e22)
- **clearance**: PCB trace or component clearance (#e74c3c)
- **schematic**: Circuit and component connections (#9b59b6)
- **manufacturing**: Manufacturing process related (#f1c40f)
- **BOM**: (Bill of Materials) Component list management (#34495e)

**Comment by Sddilora** [comment]
- **feature** (Proposal or implementation of new features) (#28a745)
- **ui/ux** (Improvements to user interface or experience) (#B32F99)
- **performance** (Enhancements to system performance) (#6f42c1)
- **security** (Identification or resolution of security concerns) (#d73a49)
- **research** (Investigation or exploration of new concepts, techniques, or approaches) (#6a737d)
- **fix** (Identification and resolution of bugs) (#2BBB8D)
- **style** (Improvements to overall readability and consistency) (#a61e4d)
- **config** (Updates or changes to system configurations) (#fd7e14)
- **monitoring** (System monitoring or metrics collection) (#DBFECD)
- **refactor** (Improvements or restructuring of existing functionality) (#bba612)
- **build** (Configuration or setup related to the build process) (#199858)
- **integration** (Tasks related to integrating with other systems or services) (#BEFD14)
- **change** (Feature changing) (#60AD72)
- **cross-team** (Collaboration required between teams) (#56AB5F)
- **not reproducible** (Issues unable to be reproduced again) (#f9d0c4)
- **parent issue** (Issue with sub-tasks) (#595068)
- **sub-issue** (Sub-task within a larger task) (#118F8E)
- **testing** (Tasks or issues for testing system functionality and reporting any identified issues) (#D3CCF8)
- **internal feedback** (Feedback identified internally) (#D7478D)
- **user feedback** (Feedback reported by users or stakeholders) (#9A5F9F)
- **planning** (Issues or tasks in the planning phase) (#d4d4d4)
- **development** (Issues or tasks currently in active development) (#2cbe4e)
- **deployment** (Issues or tasks related to deployment to production or release) (#007bff)
- **critical** (Issues requiring immediate attention) (#8b0000)
- **high priority** (Urgent issues needing immediate attention) (#C61700)
- **medium priority** (Important issues to be addressed soon) (#D93F0B)
- **low priority** (Minor issues, can be addressed later) (#FDCC21)
- **scalability** (Enhancements to system scalability and capacity) (#20B2AA)
- **compliance** (Ensuring adherence to standards and regulations) (#9370DB)
- **analysis** (Data analysis or exploration for insights) (#00FFFF)
- **deployment-strategy** (Planning and implementing deployment strategies) (#FF6347)
- **formatting** (Improvements to content readability and organization) (#6495ED)
- **configuration** (Updates or changes to system settings and configurations) (#FFA500)
- **performance-optimization** (Optimizing system performance and efficiency) (#FF69B4)
- **waiting topic** ( Issue is waiting for a topic label ) (#ffffff)

**Comment by Sddilora** [comment]
> * **feature** (Proposal or implementation of new features) (#28a745)
> * **ui/ux** (Improvements to user interface or experience) (#0366d6)
> * **performance** (Enhancements to system performance) (#6f42c1)
> * **security** (Identification or resolution of security concerns) (#d73a49)
> * **research** (Investigation or exploration of new concepts, techniques, or approaches) (#6a737d)
> * **fix** (Identification and resolution of bugs) (#d73a4a)
> * **style** (Improvements to code style or readability) (#a61e4d)
> * **config** (Updates or changes to system configurations) (#fd7e14)
> * **monitoring** (System monitoring or metrics collection) (#28a745)
> * **refactor** (Improvements or restructuring of existing functionality) (#6f42c1)
> * **build** (Configuration or setup related to the build process) (#6a737d)
> * **integration** (Tasks related to integrating with other systems or services) (#6f42c1)
> * **change** (Changes or enhancements to existing features) (#60AD72)
> * **cross team** (Collaboration required between teams) (#56AB5F)
> * **not reproducible** (Issues unable to be reproduced by team) (#f9d0c4)
> * **parent-issue** (Issue with sub-tasks) (#595068)
> * **sub-issue** (Sub-task within a larger task) (#6a737d)
> * **testing** (Tasks or issues for testing system functionality and reporting any identified issues) (#D3CCF8)
> * **internal feedback** (Feedback identified internally) (#D7478D)
> * **user feedback** (Feedback reported by users or stakeholders) (#9A5F9F)
> * **planning** (Issues or tasks in the planning phase) (#d4d4d4)
> * **development** (Issues or tasks currently in active development) (#2cbe4e)
> * **deployment** (Issues or tasks related to deployment to production or release) (#007bff)
> * **critical** (Issues requiring immediate attention) (#8b0000)
> * **high-priority** (Urgent issues needing immediate attention) (#C61700)
> * **medium-priority** (Important issues to be addressed soon) (#D93F0B)
> * **low-priority** (Minor issues, can be addressed later) (#FDCC21)
> * **scalability** (Enhancements to system scalability and capacity) (#20B2AA)
> * **compliance** (Ensuring adherence to standards and regulations) (#9370DB)
> * **analysis** (Data analysis or exploration for insights) (#00FFFF)
> * **deployment-strategy** (Planning and implementing deployment strategies) (#FF6347)
> * **formatting** (Improvements to content readability and organization) (#6495ED)
> * **configuration** (Updates or changes to system settings and configurations) (#FFA500)
> * **performance-optimization** (Optimizing system performance and efficiency) (#FF69B4)

These are the labels that are considered to be common to all repos. If there are other labels that you think will be suitable for the general public, please contact us.

**Comment by Maraza25** [comment]
**With this python script, you can add labels to your repos faster**

#### Prerequisites:
Before running this script, you need to obtain a Personal Access Token (PAT) from GitHub to perform operations via the GitHub API. This token must have the appropriate permissions to manage repositories. Below are the step-by-step instructions on how to obtain your token:

1. Log in to your GitHub account.
2. Click on your profile photo in the upper right corner and select 'Settings'.
3. From the left-hand menu, click on 'Developer settings'.
4. In the left menu, go to 'Personal access tokens'.
5. Click on 'Generate new token'.
6. Enter a name for your token and select the necessary permissions. Enable the `repo` scope to manage repositories.
7. Click on 'Generate token' to create your token.
8. Store your token in a secure place. You will not be able to see your token again; if you lose it, you will need to generate a new one.

This token will be used in the script above to perform the specified operations.


```python
import requests
import json

# Your GitHub username and repo name.
username = "octoposprime"
repo = "<your-tem's-repo-name>"
# repo = "op-fe-shared" (e.g.)
# Your access token from GitHub.
token = "your-token-here"

# Add your tags here. I added a few tags as an example.
labels = [
  {
    "label": "feature",
    "description": "Proposal or implementation of new features",
    "color": "28a745"
  },
  {
    "label": "ui/ux",
    "description": "Improvements to user interface or experience",
    "color": "B32F99"
  },
  {
    "label": "performance",
    "description": "Enhancements to system performance",
    "color": "6f42c1"
  },
  {
    "label": "security",
    "description": "Identification or resolution of security concerns",
    "color": "d73a49"
  },
  {
    "label": "research",
    "description": "Investigation or exploration of new concepts, techniques, or approaches",
    "color": "6a737d"
  },
  {
    "label": "fix",
    "description": "Identification and resolution of bugs",
    "color": "2BBB8D"
  },
  {
    "label": "style",
    "description": "Improvements to overall readability and consistency",
    "color": "a61e4d"
  },
  {
    "label": "config",
    "description": "Updates or changes to system configurations",
    "color": "fd7e14"
  },
  {
    "label": "monitoring",
    "description": "System monitoring or metrics collection",
    "color": "28a745"
  },
  {
    "label": "refactor",
    "description": "Improvements or restructuring of existing functionality",
    "color": "bba612"
  },
  {
    "label": "build",
    "description": "Configuration or setup related to the build process",
    "color": "6a737d"
  },
  {
    "label": "integration",
    "description": "Tasks related to integrating with other systems or services",
    "color": "6f42c1"
  },
  {
    "label": "change",
    "description": "Changes or enhancements to existing features",
    "color": "60AD72"
  },
  {
    "label": "cross team",
    "description": "Collaboration required between teams",
    "color": "56AB5F"
  },
  {
    "label": "not reproducible",
    "description": "Issues unable to be reproduced again",
    "color": "f9d0c4"
  },
  {
    "label": "parent issue",
    "description": "Issue with sub-tasks",
    "color": "595068"
  },
  {
    "label": "sub-issue",
    "description": "Sub-task within a larger task",
    "color": "118F8E"
  },
  {
    "label": "testing",
    "description": "Tasks or issues for testing system functionality and reporting any identified issues",
    "color": "D3CCF8"
  },
  {
    "label": "internal feedback",
    "description": "Feedback identified internally",
    "color": "D7478D"
  },
  {
    "label": "user feedback",
    "description": "Feedback reported by users or stakeholders",
    "color": "9A5F9F"
  },
  {
    "label": "planning",
    "description": "Issues or tasks in the planning phase",
    "color": "d4d4d4"
  },
  {
    "label": "development",
    "description": "Issues or tasks currently in active development",
    "color": "2cbe4e"
  },
  {
    "label": "deployment",
    "description": "Issues or tasks related to deployment to production or release",
    "color": "007bff"
  },
  {
    "label": "critical",
    "description": "Issues requiring immediate attention",
    "color": "8b0000"
  },
  {
    "label": "high priority",
    "description": "Urgent issues needing immediate attention",
    "color": "C61700"
  },
  {
    "label": "medium priority",
    "description": "Important issues to be addressed soon",
    "color": "D93F0B"
  },
  {
    "label": "low priority",
    "description": "Minor issues, can be addressed later",
    "color": "FDCC21"
  },
  {
    "label": "scalability",
    "description": "Enhancements to system scalability and capacity",
    "color": "20B2AA"
  },
  {
    "label": "compliance",
    "description": "Ensuring adherence to standards and regulations",
    "color": "9370DB"
  },
  {
    "label": "analysis",
    "description": "Data analysis or exploration for insights",
    "color": "00FFFF"
  },
  {
    "label": "deployment-strategy",
    "description": "Planning and implementing deployment strategies",
    "color": "FF6347"
  },
  {
    "label": "formatting",
    "description": "Improvements to content readability and organization",
    "color": "6495ED"
  },
  {
    "label": "configuration",
    "description": "Updates or changes to system settings and configurations",
    "color": "FFA500"
  },
  {
    "label": "performance-optimization",
    "description": "Optimizing system performance and efficiency",
    "color": "FF69B4"
  }
]


# GitHub API URL
url = f"https://api.github.com/repos/{username}/{repo}/labels"

# Process each label in a loop and add it to GitHub.
for label in labels:
    # Convert the label data to JSON format.
    data = json.dumps({
        "name": label["label"],
        "description": label["description"],
        "color": label["color"]
    })
    # Make the POST request.
    response = requests.post(url, data=data, headers={
        "Authorization": f"token {token}",
        "Accept": "application/vnd.github.v3+json"
    })
    # Check the response.
    if response.status_code == 201:
        print(f'Label added successfully: {label["label"]}')
    else:
        print(f'Error adding label: {label["label"]}. Error: {response.content}')




```

**Comment by aleyna0f** [comment]
**Below are the labels we created for the design team. These labels cover 2D and 3D in a common way. We only labeled 2D and 3D to determine which drawing program the design belongs to.**


- 3D modeling: Designing objects composed of polygons, vertices, and edges. Used for (#FF5600)

- 2D modeling: Creating flat objects using vector graphics, for designing logos and illustrations. (#BF4000)

- Animation: Bringing 3D and 2D models to life using techniques such as motion, rotation, rigging, physics, and particles. (#400036)

- Rendering: Visualizing 3D models using techniques such as lighting, shading, materials, textures, and post-processing. (#B2FF55)

- Composition: Bringing together 3D and 2D models using techniques such as scene setup, camera work, depth of field, color correction, and effects. (#FFF15C)

- Color and Texture: Specifying color samples, gradients, patterns, and textures for coloring and texture design. (#F2167D)

- Effects: Specifying the use of visual effects such as shading, contours, inner glow, textures, and transformations. (#014034)

- SVG: Requesting the storage of vector graphics such as points and lines. (#996B22)

- PNG: Requesting the storage of vector graphics such as points and lines. (#94D2E3)

- JPEG: For photographs and other realistic images. (#DBF227)

**Comment by mucahitgoncu** [comment]
Balance (Changes or fixes related to balance): (#ff6347)
Platform-specific (Platform-specific issues or improvements): (#808080)
Localization (Changes or additions related to localization): (#ffa500)
Networking (Issues or improvements related to networking): (#32cd32)
Graphics (Changes or improvements related to graphics): (#1e90ff)
Sound (Changes or fixes related to sound): (#ff69b4)
Education (Tutorial content ): (#4CAF50)

**Comment by yavuz-yusuf** [comment]
#### Here is Labels for op-em-solid Repository

- **CAD**: Computer Aided Design **(#1B4F72)**

- **3D Modeling**: Mathematical coordinate-based design of an object in three dimensions **(#F39C12)**

- **2D Modeling**: Mathematical coordinate-based design of an object in two dimensions **(#52BE80)**

- **Part Design**: design of a part of a main object **(#34495E)**

- **Mold Design**: Creating the mold that will be used to shape a desired product.**(#17202A)**

- **Simulation**: Testing and analyzing a design virtually **(#CB4335)**

- **Aesthetic**: Visual appeal and pleasing appearance of a design. **(#94CB35)**

- **Interference**: The process of checking for overlapping parts in a CAD assembly **(#5C35CB)**

- **Assembly**: process of putting together different parts to create a whole functional product. **(#CB3550)**

- **Scaling**: resizing an object proportionally in software. **(#00BA6B)**

- **Exploded**: breaking down a complex object into its individual components. **(#922B21)**
